shared/api/cluster.go

```diff
@@ -367,3 +367,70 @@ func (c *ClusterGroup) SetWritable(put ClusterGroupPut) {
 	c.Description = put.Description
 	c.Members = put.Members
 }
+
+// ClusterLink represents high-level information about a cluster link.
+//
+// swagger:model
+//
+// API extension: cluster_links.
+type ClusterLink struct {
+	// The name of the cluster
+	// Example: lxd02
+	Name string `json:"name" yaml:"name"`
+
+	// The cluster endpoint addresses
+	// Example: [10.0.0.1:8443, 10.0.0.2:8443]
+	Addresses []string `json:"addresses" yaml:"addresses"`
+
+	// Description of the cluster
+	// Example: Backup LXD cluster
+	Description string `json:"description" yaml:"description"`
+
+	// List of auth groups this cluster link belongs to
+	// Example: ["foo", "bar"]
+	Groups []string `json:"groups" yaml:"groups"`
```

Comment:
I think group membership should be managed and reported via the identities API. Maybe we can keep `Groups []string` in the POST request body but I don't think it should be in PUT/PATCH/GET.